htmd/parameterization/ffmolecule.py

```diff
@@ -94,19 +72,37 @@ def __init__(self, filename=None, name=None, rtf=None, prm=None, netcharge=None,
             # If the user has specified explicit RTF and PRM files go ahead and load those
             self._rtf = RTF(rtf)
             self._prm = PRM(prm)
+        elif method == FFTypeMethod.NONE:
+            # Don't assign any atom types
+            pass
         else:
             # Otherwise make atom types using the specified method
-            # (Right now only MATCH)
             fftype = FFType(self, method=self.method)
             self._rtf = fftype._rtf
             self._prm = fftype._prm
-        if not self._rtf or not self._prm:
-            raise ValueError("RTF and PRM not defined")
 
-        self.impropers = np.array(self._rtf.impropers)
+        self.qm = qm if qm else Psi4()
+
+        self.theory_name = self.qm.theory
+        self.basis_name = self.qm.basis
+        self.basis_name = re.sub('\+', 'plus', self.basis_name)  # Replace '+' with 'plus'
```

Comment:
Hum, so you are doing this (the star and plus) at this level... Any particular reason? For me, it seems to make it more complex, but maybe I'm wrong

Reply:
QM classes can write the result to any directory, not necessary named after method and basis. I don't see the benefit to hard code this.

Reply:
this is/was being done at the client level (where the outdir is/was based on the conjunction of the several options), in order for the directories to work in Unix.
What I don't understand is the need to do this (change * to star) other than that.

Reply:
Sorry, this is being done at the class level rather. My idea was that this could be moved to the `output_directory_name` method instead of being here.